src/pages/superadmin/tableComponent/__tests__/TableComponent.spec.tsx

```diff
@@ -423,4 +423,64 @@ describe('MyTable Component', () => {
     const paidBounties = getAllByText('paid');
     expect(paidBounties.length).toBe(1);
   });
+
+  it('simulates filtering bounties by status: open, assigned, paid', async () => {
+    let filteredBounties = mockBounties.filter((bounty: any) => bounty.status === 'open');
+    const { rerender } = render(<MyTable {...MockTableProps} bounties={filteredBounties} />);
+    expect(screen.getByText('Bounty 1')).toBeInTheDocument();
+    expect(screen.queryByText('Bounty 2')).not.toBeInTheDocument();
+    expect(screen.queryByText('Bounty 3')).not.toBeInTheDocument();
+
+    filteredBounties = mockBounties.filter((bounty: any) => bounty.status === 'assigned');
+    rerender(<MyTable {...MockTableProps} bounties={filteredBounties} />);
+    expect(screen.getByText('Bounty 2')).toBeInTheDocument();
+    expect(screen.queryByText('Bounty 1')).not.toBeInTheDocument();
+    expect(screen.queryByText('Bounty 3')).not.toBeInTheDocument();
+
+    filteredBounties = mockBounties.filter((bounty: any) => bounty.status === 'paid');
+    rerender(<MyTable {...MockTableProps} bounties={filteredBounties} />);
+    expect(screen.getByText('Bounty 3')).toBeInTheDocument();
+    expect(screen.queryByText('Bounty 1')).not.toBeInTheDocument();
+    expect(screen.queryByText('Bounty 2')).not.toBeInTheDocument();
+  });
+
+  it('sorts the bounties according to admin select Newest', async () => {
```

Comment:
@aliraza556 you should be mocking that a call is made to the backend when the sort direction is changed, because the backend handles the sorting

Reply:
I don't think we need to mock the API call cause this component directly takes in bounties which we can mock directly with props. It seems like bounties is only modified when we change `currentPage` or use `getBounties()`.